LambdaS3FileZipper/Exceptions/ResourceNotFoundException.cs

```diff
@@ -4,7 +4,8 @@ namespace LambdaS3FileZipper.Exceptions
 {
     public class ResourceNotFoundException : ArgumentException
     {
-        public ResourceNotFoundException(string bucket, string resource) : base($"No files found under {bucket}:{resource}")
+        public ResourceNotFoundException(string bucket, string resource, string resourceExpressionPattern)
+            : base($"No files found for resource `{bucket}/{resource}` that match expression `{resourceExpressionPattern}`")
```

Comment:
Please remove the string interpolation from the exception messages and expose those via properties.  This ends up making a mess of most logging systems that key off of exception message.

Reply:
Do you have an example pattern that I can use for exposing properties on exceptions? Not really sure what you have in mind 😅 .

Reply:
How about?
```
public class ResourceNotFoundException : ArgumentException
    {
		public string Bucket { get; set; }
		public string Resource { get; set; }
		public string ResourceExpressionPattern { get; set; } 
		
        public ResourceNotFoundException(string bucket, string resource, string resourceExpressionPattern)
            : base($"No files found matching the expression")
		{
			Bucket = bucket;
			Resource = resource;
			ResourceExpressionPattern = resourceExpressionPattern;
        }
    }
```
Have the error message string mutate can cause problems with systems like Seq where they key off of the hash of a string.

Reply:
> Have the error message string mutate can cause problems with systems like Seq where they key off of the hash of a string.

That's the _key_ right there 😄 . I'll make the changes.

On a related note, do you happen to know if those properties get serialized as is?